agent/session.go

```diff
@@ -217,6 +223,58 @@ func (s *session) handleSessionMessage(ctx context.Context, msg *api.SessionMess
 	}
 }
 
+func (s *session) taskExecutions(ctx context.Context) error {
+	// TODO(fntlnz): REMOVE THE CLIENT HERE find a way to have here the docker client instantiated in cmd/swarmd
+	// instead of creating a new one or at least use the same creation logic bcause the other spports containerd
+	dclient, err := engineapi.NewEnvClient()
+	client := api.NewTaskExecClient(s.conn.ClientConn)
+	attachment, err := client.Attach(ctx)
+	if err != nil {
+		return err
+	}
+
+	// TODO(fntlnz): rework this to support streaming over the same exec and not just sending/receiveing
+	// TODO(fntlnz): pass out errors instead of just conintuing
+
+	for {
+		data, err := attachment.Recv()
+
+		if err != nil {
+			return err
+		}
+		execCfg := types.ExecConfig{
+			User:         "root",
+			AttachStdout: true,
+			AttachStderr: true,
+			AttachStdin:  true,
+			Cmd:          strings.Split(string(data.Message), " "),
```

Comment:
I would look into making this an explicit protobuf field in the exec flow instead of reusing `Message`.